front/components/vaults/EditVaultManagedDatasourcesViews.tsx

```diff
@@ -0,0 +1,162 @@
+import { Button, PlusIcon, Spinner } from "@dust-tt/sparkle";
+import type {
+  APIError,
+  ManagedDataSourceViewsSelectedNodes,
+  VaultType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import React, { useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import VaultManagedDataSourcesViewsModal from "@app/components/vaults/VaultManagedDatasourcesViewsModal";
+import { useVaultDataSourceViews } from "@app/lib/swr";
+
+export function EditVaultManagedDataSourcesViews({
+  owner,
+  vault,
+  systemVault,
+}: {
+  owner: WorkspaceType;
+  vault: VaultType;
+  systemVault: VaultType;
+}) {
+  const sendNotification = React.useContext(SendNotificationsContext);
+
+  const [showDataSourcesModal, setShowDataSourcesModal] = useState(false);
+
+  // DataSources Views of the current vault.
+  const {
+    vaultDataSourceViews,
+    isVaultDataSourceViewsLoading,
+    mutateVaultDataSourceViews,
+  } = useVaultDataSourceViews({
+    workspaceId: owner.sId,
+    vaultId: vault.sId,
+    category: "managed",
+  });
+
+  // DataSources Views of the system vault holding the managed datasources we want to select data from.
+  const {
+    vaultDataSourceViews: systemVaultDataSourceViews,
+    isVaultDataSourceViewsLoading: isSystemVaultDataSourceViewsLoading,
+  } = useVaultDataSourceViews({
+    workspaceId: owner.sId,
+    vaultId: systemVault.sId,
+    category: "managed",
+  });
+
+  const updateVaultDataSourceViews = async (
+    selectedNodes: ManagedDataSourceViewsSelectedNodes
+  ) => {
+    let error = null;
+    await Promise.all(
+      selectedNodes.map(async (sDs) => {
```

Comment:
So here, a node is always a top level data source view? Or it can be content node?

Reply:
yes datasourceview